packages/angular/pwa/pwa/index.ts

```diff
@@ -70,43 +59,32 @@ function updateIndexFile(options: PwaOptions): Rule {
     const content = buffer.toString();
     const lines = content.split('\n');
     let closingHeadTagLineIndex = -1;
-    let closingHeadTagLine = '';
     let closingBodyTagLineIndex = -1;
-    let closingBodyTagLine = '';
-    lines.forEach((line: string, index: number) => {
-      if (/<\/head>/.test(line) && closingHeadTagLineIndex === -1) {
-        closingHeadTagLine = line;
+    lines.forEach((line, index) => {
+      if (closingHeadTagLineIndex === -1 && /<\/head>/.test(line)) {
         closingHeadTagLineIndex = index;
-      }
-
-      if (/<\/body>/.test(line) && closingBodyTagLineIndex === -1) {
-        closingBodyTagLine = line;
+      } else if (closingBodyTagLineIndex === -1 && /<\/body>/.test(line)) {
         closingBodyTagLineIndex = index;
       }
     });
 
-    const headTagIndent = getIndent(closingHeadTagLine) + '  ';
+    const headIndent = getIndent(lines[closingHeadTagLineIndex]) + '  ';
     const itemsToAddToHead = [
       '<link rel="manifest" href="manifest.json">',
       '<meta name="theme-color" content="#1976d2">',
     ];
 
-    const textToInsertIntoHead = itemsToAddToHead
-      .map(text => headTagIndent + text)
-      .join('\n');
-
-    const bodyTagIndent = getIndent(closingBodyTagLine) + '  ';
-    const itemsToAddToBody
-      = '<noscript>Please enable JavaScript to continue using this application.</noscript>';
-
-    const textToInsertIntoBody = bodyTagIndent + itemsToAddToBody;
+    const bodyIndent = getIndent(lines[closingBodyTagLineIndex]) + '  ';
+    const itemsToAddToBody = [
+      '<noscript>Please enable JavaScript to continue using this application.</noscript>',
```

Comment:
Adding this to the actual template `index.html` may make sense since it applies to a standard web application as well.  Thoughts?

Reply:
True. I guess this was added as poart of `@angular/pwa`, because good PWA practices include having fallback content for when JS is disabled, but this is indeed useful for non-PWA apps as well.
Should Imove it to `index.html`?

Reply:
I think it eventually should but probably should be left to a separate PR.  However, the addition here would need to stay either way to support projects that didn't get created with the new `index.html`.  The code here should be augment now though to check if the `noscript`element is already present and only add if not.  Since someone could have added the element manually as well.

Reply:
Actually, I just realized that a proper solution should also check whether `link[rel="manifest"]` and `meta[name="theme-color"]` already exists as well. Since this PR is only refactoring that part of the code (no behavior changes), I'd rather leave the checks for another PR.

Reply:
Good point and works for me.